common/src/main/java/com/wynntils/models/mobtotem/MobTotemModel.java

```diff
@@ -0,0 +1,130 @@
+    private static final double TOTEM_COORDINATE_DIFFERENCE = 0.2d;
+
+    private final HashMap<Integer, MobTotem> mobTotems = new HashMap<>();
+
+    public MobTotemModel(WorldStateModel worldState) {
+        super(List.of(worldState));
+    }
+
+    @SubscribeEvent
+    public void onTotemRename(EntityLabelChangedEvent e) {
+        if (!Models.WorldState.onWorld()) return;
+
+        Entity entity = e.getEntity();
+        if (!(entity instanceof ArmorStand as)) return;
+
+        // If a new mob totem just appeared, add it to the unstarted list
+        // Totem timers do not match the MOB_TOTEM_NAME pattern
+        Matcher nameMatcher = MOB_TOTEM_NAME.matcher(e.getName());
+        if (nameMatcher.find()) {
+            int mobTotemId = e.getEntity().getId();
+
+            if (mobTotems.containsKey(mobTotemId)) return; // If the totem is already in the list, don't add it again
+
+            mobTotems.put(mobTotemId, new MobTotem(new Location(as), nameMatcher.group(1)));
+            return;
+        }
+
+        if (mobTotems.isEmpty()) return;
+
+        for (MobTotem mobTotem : mobTotems.values()) {
+            // Exact equality is fine here because the totem is stationary
+            if (as.getX() == mobTotem.getLocation().x()
+                    && as.getY() == (mobTotem.getLocation().y() + TOTEM_COORDINATE_DIFFERENCE)
+                    && as.getZ() == mobTotem.getLocation().z()) {
+                Matcher timerMatcher = MOB_TOTEM_TIMER.matcher(e.getName());
+                if (timerMatcher.find()) {
+                    mobTotem.setTimerString(timerMatcher.group(1));
+                    return;
+                }
+            }
+        }
+    }
+
+    @SubscribeEvent
+    public void onTotemDestroy(RemoveEntitiesEvent e) {
+        if (!Models.WorldState.onWorld()) return;
+
+        e.getEntityIds().forEach(mobTotems::remove);
+    }
+
+    public List<MobTotem> getMobTotems() {
+        return mobTotems.values().stream().toList();
+    }
+
+    public static class MobTotem {
+        private final Location location;
+        private final String owner;
+        private String timerString;
+
+        public MobTotem(Location location, String owner) {
+            this.location = location;
+            this.owner = owner;
+        }
+
+        public Location getLocation() {
+            return location;
+        }
+
+        public String getOwner() {
+            return owner;
+        }
+
+        public String getTimerString() {
+            return timerString;
+        }
+
+        public void setTimerString(String timerString) {
+            this.timerString = timerString;
+        }
+
+        public int getDistanceToPlayer() {
+            // y() - 4 because the entity is 4 blocks above the ground
+            return (int)
+                    Math.sqrt(McUtils.player().distanceToSqr(new Vec3(location.x(), location.y() - 4, location.z())));
+        }
+
+        public double getLookAngleDiff() {
```

Comment:
I'm guessing this is not needed right now. It's up to you if you want to keep it anyway.

Reply:
Would like to keep it for in the future